### PR TITLE
[MM-52710] Adding a separator to the channel search string

### PR DESCRIPTION
**Pull request by BenCookie95**
#### Summary
Not a pretty solution but it works the best. I could separate the search strings into a string array and then filter through each item and compare against `channelPrefix` but this way is more efficient since it's just one `.includes()`.

The original problem was that when you search for something it could include some confusing results due to the different fields being concatenated together.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-52710

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/suggestion/switch_channel_provider.tsx [hunk 1]
```diff
@@ -408,7 +409,7 @@ function makeChannelSearchFilter(channelPrefix: string) {
                         continue;
                     }
                     const {nickname, username} = user;
-                    userString = `${nickname}${username}${Utils.getFullName(user)}`;
+                    userString = `${nickname}${SEPARATOR}${username}${SEPARATOR}${Utils.getFullName(user)}`;
```
Comment (tasawar-hussain) [inline webapp/channels/src/components/suggestion/switch_channel_provider.tsx:412]: more clean
```suggestion
                    userString = [nickname, username, Utils.getFullName(user)].join(SEPARATOR);
```

**Review by tasawar-hussain — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/suggestion/switch_channel_provider.tsx (hunk 1 above)
Comment (BenCookie95) [reply webapp/channels/src/components/suggestion/switch_channel_provider.tsx]: Thanks for the suggestion @tasawar-hussain

**Review by tasawar-hussain — APPROVED** [review]

**Review by pvev — APPROVED** [review]
Code LGTM! validated and worked as expected. Thanks @BenCookie95 👍

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Tested to ensure the specific issue is resolved
- Smoke tested the channel switcher search
- No relevant records ever seem to be missing 
- No issues found

LGTM!

**Pull request merged**